### ISSUE TITLE
Azure OpenAI: Re-enable project-reference tests

### ISSUE THREAD

**Issue by annelo-msft** [Azure/azure-sdk-for-net#44899]
Project reference tests are disabled in this PR to unblock Core PRs while SCM and 3p OAI are in preview: https://github.com/Azure/azure-sdk-for-net/pull/44896

Once these libraries have GA'ed and we are no longer taking breaking changes, we should re-enable these pipelines.

Will also need to back out this change: https://github.com/Azure/azure-sdk-for-net/pull/44825/commits/a312bec7c7e42566f4155e173d08f0da726edd34

**Comment by annelo-msft** [comment]
Closing in favor of https://github.com/Azure/azure-sdk-for-net/issues/45996
